pkg/didcomm/protocol/transport/http/http_transport.go

```diff
@@ -0,0 +1,262 @@
+func newHTTPClient(cfg *OutboundCommConfig) (*http.Client, error) {
+	var err error
+	var caCertPool tlsCertPool.CertPool
+	if cfg.CACertsPaths != "" {
+		caCertPool, err = tlsCertPool.NewCertPool(false)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to create new Cert Pool")
+		}
+
+		caCertsPaths := strings.Split(cfg.CACertsPaths, ",")
+		var caCerts []string
+		for _, path := range caCertsPaths {
+			if path == "" {
+				continue
+			}
+			// Create a pool with server certificates
+			caCert, e := ioutil.ReadFile(filepath.Clean(path))
+			if e != nil {
+				return nil, errors.Wrap(e, "Failed Reading server certificate")
+			}
+			caCerts = append(caCerts, string(caCert))
+		}
+
+		caCertPool.Add(tlsCertPool.DecodeCerts(caCerts)...)
+	} else {
+		caCertPool, err = tlsCertPool.NewCertPool(true)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// update the config's caCertPool
+	cfg.caCertPool = caCertPool
+
+	tlsConfig, err := buildNewCertPool(cfg.caCertPool)
+	if err != nil {
+		log.Printf("HTTP Transport - Failed to build/get Cert Pool: %s", err)
+		return nil, err
+	}
+
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConfig,
+		},
+		Timeout: cfg.Timeout,
+	}, nil
+}
+
+func buildNewCertPool(tlsCertPool tlsCertPool.CertPool) (*tls.Config, error) {
+	cp, err := tlsCertPool.Get()
+	if err != nil {
+		return nil, err
+	}
+	// Create TLS configuration
+	// add other agents (server) certs here.. for now we trust CA certs
+	// Since this config cannot be manipulated once the client is created,
+	// we must find a way to trust other agents here (or only trust root CAs).
+	tlsConfig := &tls.Config{
+		// if RootCAs is nil, client will use the host's root CA instead
+		RootCAs:      cp,
+		Certificates: nil,
+	}
+	return tlsConfig, nil
+}
+
+// DIDCommRequestHandler will create a new handler to enforce Did-Comm HTTP transport specs
+// then routes processing to the passed in handler argument
+func DIDCommRequestHandler(handler http.Handler, commHandler *transport.DIDCommHandler) http.Handler {
+
+	validateHandler(commHandler)
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case commHandler.ExchangeRequest.Path:
+			processPOSTRequest(w, r, commHandler.ExchangeRequest.HandlerFunc)
+			return
+		case commHandler.ExchangeResponse.Path:
+			processPOSTRequest(w, r, commHandler.ExchangeResponse.HandlerFunc)
+			return
+		case commHandler.IntroductionProposal.Path:
+			processPOSTRequest(w, r, commHandler.IntroductionProposal.HandlerFunc)
+			return
+		case commHandler.IntroductionRequest.Path:
+			processPOSTRequest(w, r, commHandler.IntroductionRequest.HandlerFunc)
+			return
+		case commHandler.IntroductionResponse.Path:
+			processPOSTRequest(w, r, commHandler.IntroductionResponse.HandlerFunc)
+			return
+		}
+
+		handler.ServeHTTP(w, r)
+	})
+}
+
+// TODO Log error message with common trustbloc/logger-lib
```

Comment:
This comment will need to be updated to work within Hyperledger.

Reply:
will update comment..